### PR TITLE
Document the use of HTTP/1 full duplex support

### PR DESCRIPTION
**Pull request by skonto**
## Proposed Changes 

- Documents https://github.com/knative/serving/issues/12387. Some similar docs from another project [here](https://caddyserver.com/docs/json/apps/http/servers/enable_full_duplex/).
- Adds proper warning for cases where clients may not support the feature.

### CODE REVIEW

File: docs/serving/services/http-protocol.md [hunk 1]
```diff
@@ -50,3 +52,31 @@ You can override the default behavior for each Service or global configuration.
...
+  namespace: default
+  annotations:
+    features.knative.dev/http-full-duplex: "Enabled"
+spec:
+  template:
+    spec:
+...
+```
+
+!!! warning
```
Comment (ReToCode) [inline docs/serving/services/http-protocol.md:80]: I probably would move that to the top of the YAML. Other than that LGTM

File: docs/serving/services/http-protocol.md [hunk 2]
```diff
@@ -50,3 +52,31 @@ You can override the default behavior for each Service or global configuration.
...
+In order to enable the HTTP/1 full duplex support you can set the corresponding annotation at the service level as follows:
+
+```yaml
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: example-service
+  namespace: default
+  annotations:
+    features.knative.dev/http-full-duplex: "Enabled"
```
Comment (dprotaso) [inline docs/serving/services/http-protocol.md:73]: does the casing matter? eg. I assume "enabled" works?

Comment (ReToCode) [reply docs/serving/services/http-protocol.md:73]: I haven't tested this specific case, but usually all our flags are string.EqualFold'ded.

**Review by ReToCode — APPROVED** [review]
/lgtm
/approve

**Pull request merged**